File: typescript/src/repl.ts
import { Token, Tokenizer } from "./lexer";

import readline from "readline";

const rs = readline.createInterface({
    input: process.stdin,
});

rs.on("line", (input) => {
    const tokenizer = new Tokenizer(input);

    while (true) {
        const token = tokenizer.getNextToken()
        console.log(token);
        if (token.type === "EOF") {
            break;
        }
    }
});

rs.on("close", () => {
    console.log("suck it chat");
});

File: perl/repl.pl
use Term::ReadLine;
use lib 'Lexer';
use Lexer;
use Token;

my $term = Term::ReadLine->new('MonkeyLang Token REPL');

# Disable prompt underline
$term->ornaments(0); 

while (defined(my $input = $term->readline('> '))) {
    my $lexer = Lexer->new($input);

    my $token = $lexer->nextToken();
    while($token->getType() ne Token->EOF) {
        my $tokenType = $token->getType()->getName();
        my $lit = $token->getLiteral();
        print "$tokenType: $lit";
        print "\n";
        $token = $lexer->nextToken();
    }
}

